js/scripts.js

```diff
@@ -190,40 +171,108 @@ let pokemonRepository = (function() {
         button.setAttribute('data-toggle', 'modal');
         button.setAttribute('data-target', '#pokemodal');
 
+        // Create flex element to hold name and id elements
+        let buttonTextWrapper = document.createElement('div');
+        buttonTextWrapper.classList.add('pokemon-list__pokemon-card--text');
+
+        // Create name element to include in button
+        let buttonName = document.createElement('p');
+        buttonName.innerText = `${pokemon.name}`;
+
+        // Create image element to include in button
+        let buttonId = document.createElement('p');
+        buttonId.innerText = `#${pokemon.id}`;
+
+        // Create image element to include in button
+        let buttonImage = document.createElement('img');
+        buttonImage.setAttribute('src', pokemon.imageUrl);
+
         // Add click event to button to show item details.
         addEvent(button, pokemon);
 
+        buttonTextWrapper.appendChild(buttonName);
+        buttonTextWrapper.appendChild(buttonId);
+        button.appendChild(buttonTextWrapper);
+        button.appendChild(buttonImage);
         htmlList.appendChild(button);
     }
 
+  // Search through pokémon by item name
+  let searchValue = document.getElementById('searchBar');
+
+  searchValue.addEventListener('keyup', function (e) {
+    let searchString = e.target.value.toLowerCase();
+
+    let itemsToHide = pokemonList.filter(function (item) {
+      // Find non-matching names
+      if (
+        !item.name.toLowerCase().includes(searchString)
+      ) {
+        return item;
+      }
+    });
+    let itemsToShow = pokemonList.filter(function (item) {
+      // Find matching names
+      if (
+        item.name.toLowerCase().includes(searchString)
+      ) {
+        return item;
+      }
+    });
+
+    itemsToHide.map((item) => {
+      document.getElementById(item.name).classList.add('d-none');
+    });
+    itemsToShow.map((item) => {
+      document.getElementById(item.name).classList.remove('d-none');
+    });
+  });
+
+    // Toggle search bar by clicking the search icon
+    let searchButton = document.querySelector('.search-button');
+    $(document).ready(function () {
+        $(searchButton).click(function () {
+            $('.search-bar').toggleClass('d-none');
+        });
+    });
+
     return {
         addListItem,
         loadList,
-        getAll
+        getAll,
+        loadDetails,
+        showLoadingMessage,
+        hideLoadingMessage
     };
 })();
 
-// Writes the names of each pokémon to the DOM
+// Load all the data and create the card list
+pokemonRepository.showLoadingMessage();
 pokemonRepository.loadList().then(function () {
+    pokemonRepository.hideLoadingMessage();
     pokemonRepository.getAll().forEach(function(pokemon) {
 
-        pokemonRepository.addListItem(pokemon);
+        pokemonRepository.loadDetails(pokemon)
+        .then(function() {
+            pokemonRepository.addListItem(pokemon)
+        });
```

Comment:
Let us instead call the `loadDetails` function inside the definition of `addListItem` function. In that way they should load as required.
Sample of what I tried.
<img width="1398" alt="Screenshot 2022-10-10 at 17 50 25" src="https://user-images.githubusercontent.com/7220167/194894791-5fa13222-4f4e-464f-8859-30f2dee5d960.png">